knowledge-base/grid-kb-style-filtered-columns.md

```diff
@@ -0,0 +1,151 @@
+---
+
+## Environment
+
+<table>
+    <tbody>
+	    <tr>
+	    	<td>Product</td>
+	    	<td>Grid for Blazor</td>
+	    </tr>
+    </tbody>
+</table>
+
+## Description
+
+I want to style filtered columns in the Grid for Blazor using RGBA for better visibility. Specifically, when a filter is applied, I want the background color of column cells to change.
+
+This knowledge base article answers the following questions:
+- How to apply custom styles to filtered columns in a Blazor Grid?
+- How to enhance the visibility of filtered columns in Blazor Grid using CSS?
+
+## Solution
+
+To style filtered columns in a Telerik Grid for Blazor: 
+1. Utilize the [`OnCellRender`]({%slug grid-column-events%}#oncellrender) and [`OnStateChanged`]({%slug grid-state%}#onstatechanged) events 
+2. Apply a custom CSS class to the filtered column when a filter is active
+
+>caption Grid with styled filtered column.
+
+`````CSHTML
+@using Telerik.DataSource.Extensions
+@using Telerik.DataSource
+
+<style>
+    .highlighted-column {
+        background-color: rgba(255, 0, 0, 0.2); /* Change to your desired RGBA */
+        color: #191970;
+    }
+</style>
+
+<TelerikGrid TItem="@Employee" 
+             OnRead="@ReadItems"
+             FilterMode="@GridFilterMode.FilterRow"
+             Sortable="true" 
+             Pageable="true" 
+             OnStateChanged="@OnGridStateChanged">
+    <GridColumns>
+        <GridColumn Field="@nameof(Employee.ID)" Filterable="false" Title="ID" />
+        <GridColumn Field="@nameof(Employee.Name)" OnCellRender="@OnNameColumnCellRender" Title="Name" />
+        <GridColumn Field="@nameof(Employee.HireDate)" OnCellRender="@OnHireDateColumnCellRender" Title="Hire Date" />
+    </GridColumns>
+</TelerikGrid>
+
+@code {
+    private List<Employee> SourceData { get; set; }
+    private string FilteredColumn { get; set; }
+
+    private void OnNameColumnCellRender(GridCellRenderEventArgs args)
+    {
+        if (FilteredColumn == nameof(Employee.Name))
+        {
+            args.Class = "highlighted-column"; // Apply the custom CSS class
+        }
+    }
+
+    private void OnHireDateColumnCellRender(GridCellRenderEventArgs args)
+    {
+        if (FilteredColumn == nameof(Employee.HireDate))
+        {
+            args.Class = "highlighted-column"; // Apply the custom CSS class
+        }
+    }
+
+    private async Task OnGridStateChanged(GridStateEventArgs<Employee> args)
+    {
+        if (args.PropertyName == "FilterDescriptors")
+        {
+            if (args.GridState.FilterDescriptors.Any())
+            {
+                var filterDescriptors = new List<IFilterDescriptor>(args.GridState.FilterDescriptors);
+                var filterDescriptor = filterDescriptors[0] as CompositeFilterDescriptor;
+                var filteredColumn = filterDescriptor.FilterDescriptors[0] as FilterDescriptor;
+
+                // Capture the filtered column's field name
+                FilteredColumn = filteredColumn.Member;
```

Comment:
This logic supports highlighting for just one column (the leftmost one), while the Grid can have many.